pallets/creditcoin/src/tests.rs

```diff
@@ -2215,3 +2235,188 @@ fn exempt_should_succeed() {
+	});
+}
+
+#[test]
+fn on_initialize_removes_expired_deals_without_transfers() {
+	ExtBuilder::default().build_offchain_and_execute(|| {
+		System::set_block_number(1);
+		let mut expected_deal_orders = Vec::new();
+
+		let now = System::block_number();
+		for expiration_block in now..=20 {
+			let address1 =
+				format!("0x{:02}F03B407c01E7cD3CBea99509d93f8DDDC8C6FB", expiration_block.clone())
+					.hex_to_address();
+			let address2 =
+				format!("0x{:02}220A0cf47c7B9Be7A2E6BA89F429762e7b9aDb", expiration_block.clone())
+					.hex_to_address();
+
+			let test_info = TestInfo {
+				lender: RegisteredAddress::new(address1, 111, Blockchain::Rinkeby),
+				borrower: RegisteredAddress::new(address2, 222, Blockchain::Rinkeby),
+				blockchain: Blockchain::Rinkeby,
+				loan_terms: LoanTerms {
+					amount: 2_000_000u64.into(),
+					interest_rate: Default::default(),
+					term_length: Duration::from_millis(1_000_000),
+				},
+				ask_guid: format!("{:?}-ask-guid", expiration_block.clone())
+					.as_bytes()
+					.into_bounded(),
+				bid_guid: format!("{:?}-bid-guid", expiration_block.clone())
+					.as_bytes()
+					.into_bounded(),
+				expiration_block: expiration_block.clone(),
+			};
+
+			let (_, offer_id) = test_info.create_offer();
+			assert_ok!(Creditcoin::add_deal_order(
+				Origin::signed(test_info.borrower.account_id.clone()),
+				offer_id.clone(),
+				expiration_block.clone(),
+			));
+			let deal_order_id = DealOrderId::new::<Test>(expiration_block.clone(), &offer_id);
+			let deal_order =
+				Creditcoin::deal_orders(deal_order_id.expiration(), deal_order_id.hash()).unwrap();
+
+			// fund only deal orders which expire at even blocks
+			if expiration_block % 2 == 0 {
+				let tx = format!("0xfafafa{:02}", expiration_block.clone());
+				assert_ok!(Creditcoin::register_funding_transfer(
+					Origin::signed(test_info.lender.account_id.clone()),
+					TransferKind::Native,
+					deal_order_id.clone(),
+					tx.as_bytes().into_bounded()
+				));
+				test_info.mock_transfer(
+					&test_info.lender,
+					&test_info.borrower,
+					deal_order.terms.amount,
+					&deal_order_id,
+					tx,
+				);
+
+				// it's funded so it should be kept
+				expected_deal_orders.push(deal_order_id.clone());
+				log::debug!(
+					"****** DEBUG: EXP BLOCK={:?}, expected={:?}",
+					expiration_block.clone(),
+					deal_order_id.clone()
+				);
+			} else if expiration_block > 15 {
+				// still hasn't expired so should be kept regardless
+				expected_deal_orders.push(deal_order_id.clone());
+				log::debug!(
+					"****** DEBUG: EXP BLOCK={:?}, expected={:?}",
+					expiration_block.clone(),
+					deal_order_id.clone()
+				);
+			}
+		}
+
+		// advance blocks, will perform housekeeping
+		roll_to(15);
+
+		for expected_order_id in expected_deal_orders.iter() {
+			log::debug!("***** DEBUG: TRYING TO FIND EXPECTED {:?}", expected_order_id);
+			let _order = DealOrders::<Test>::try_get_id(&expected_order_id).unwrap();
```

Comment:
@nathanwhit I get this error here:

```
2022-04-15T19:34:04.713230Z DEBUG pallet_creditcoin::tests: ***** DEBUG: TRYING TO FIND EXPECTED DealOrderId(2, 0x65fd06b8119c6515800def0f7f75163a766e2ccd00f8c215f62940db894e2405)    
2022-04-15T19:34:04.713361Z TRACE get_version_1: state: method="Get" ext_id=9947 key=d766358cca00233e6155d7c14e2c085fa6adf9ceadfc4cfb4dbc98b9181657e6b02de844403ec7ea020000000000000065fd06b8119c6515800def0f7f75163a766e2ccd00f8c215f62940db894e2405 result=None result_encoded=00
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ()', pallets/creditcoin/src/tests.rs:2419:77
```

I want to inspect that all funded deal orders are still present. Any idea what I am doing wrong ?

Reply:
You have to call `fund_deal_order` after registering the funding transfer in order to attach the transfer to the deal order.